src/openssl.c

```diff
@@ -780,43 +1034,129 @@ _libssh2_sha256(const unsigned char *message, unsigned long len,
 }
 
 int
-_libssh2_md5_init(libssh2_md5_ctx *ctx)
+_libssh2_sha384_init(libssh2_sha384_ctx *ctx)
 {
 #ifdef HAVE_OPAQUE_STRUCTS
     *ctx = EVP_MD_CTX_new();
 
     if (*ctx == NULL)
-        return 0;
+    return 0;
 
-    if (EVP_DigestInit(*ctx, EVP_get_digestbyname("md5")))
-        return 1;
+    if (EVP_DigestInit(*ctx, EVP_get_digestbyname("sha384")))
+    return 1;
 
     EVP_MD_CTX_free(*ctx);
     *ctx = NULL;
 
     return 0;
 #else
     EVP_MD_CTX_init(ctx);
-    return EVP_DigestInit(ctx, EVP_get_digestbyname("md5"));
+    return EVP_DigestInit(ctx, EVP_get_digestbyname("sha384"));
 #endif
 }
 
-static unsigned char *
-write_bn(unsigned char *buf, const BIGNUM *bn, int bn_bytes)
+int
+_libssh2_sha384(const unsigned char *message, unsigned long len,
+    unsigned char *out)
 {
-    unsigned char *p = buf;
+#ifdef HAVE_OPAQUE_STRUCTS
+    EVP_MD_CTX * ctx = EVP_MD_CTX_new();
 
-    /* Left space for bn size which will be written below. */
-    p += 4;
+    if (ctx == NULL)
+    return 1; /* error */
```

Comment:
The missing indentation issue is present here and in a bunch of further blocks later in the new code.